.github/workflows/test.yml

```diff
@@ -0,0 +1,49 @@
+name: CI testing
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  serialization-test:
+    runs-on: ubuntu-latest
+
+    strategy:
+        fail-fast: false # run all tests even if one fails
+        matrix:
+            python-version: [ "3.8", "3.9", "3.10", "3.11"]
+            input-file: [ "millipede_wilks/run00136662.evt000035405932.neutrino_1" ]
+
+    steps:
+      - name: Checkout main repository
+        uses: actions/checkout@v3
+
+      - name: Checkout test data from skymap_scanner
+        uses: actions/checkout@v3
+        with:
+          repository: icecube/skymap_scanner
+          ref: main
+          path: skymap_scanner
+          sparse-checkout: tests/data
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install
+        run: |
+          pip install .
+
+      - name: Run conversion script
+        # working-directory: auxiliary-repo/path/to/directory  # Replace 'path/to/directory' with the path to the directory containing your Python script
+        run: python examples/npz_to_json.py --input-file skymap_scanner/tests/data/results_npz/${{ matrix.input-file }}.npz --output-dir .
+    
+      # Wait for skymap_scanner PR#163
+      # - name: Compare output
+      #   run: |
+      #       tree
+      #       outfile=$(ls -Art | tail -n 1)
+      #       diff $outfile skymap_scanner/tests/data/results_json/${{ matrix.input-file }}.json
```

Comment:
for the `python-version` stuff feel free to follow this as an example, it follows the CI ecosystem we've built for wipac packages: https://github.com/icecube/skyreader/blob/main/.github/workflows/wipac-cicd.yml#L61-L86

This will automatically increment versions when new Python releases are made and/or drop no longer supported ones.

On second thought, you may want to combine your tests into `wipac-cicd.yml`. That would also allow you to make the tests a dependency for the `release` step

Reply:
Thanks, it makes sense.